File: assignment4/comp3007_f18_101037159_a4_2a.pl
% Owen Craston
%  101037159
%  comp 3007 assignment 4/4

% base case
countInRange([], _, _, 0).
% in range
countInRange([H|T], Min, Max, Count) :- (H >= Min, H =< Max), countInRange(T, Min, Max, Count1), Count is 1 + Count1.
% out of range
countInRange([H|T], Min, Max, Count) :- (H < Min ; H > Max), countInRange(T, Min, Max, Count).

File: assignment4/comp3007_f18_101037159_a4_2a.hs
-- Owen Craston
-- 101037159
-- assignment 4/4


countInRange :: [Int] -> Int -> Int -> Int
-- pass in value and start couneter at 0
countInRange l min max = countInRange' l min max 0


countInRange' :: [Int] -> Int -> Int -> Int -> Int
countInRange' [] _ _ count  = count
countInRange' (h:t) minimum maximum count
     | h >= minimum && h <= maximum = countInRange' t minimum maximum (count+1)
     | otherwise = countInRange' t minimum maximum (count)
